compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4895,19 +4895,22 @@ object Types {
     /** Instantiate variable with given type */
     def instantiateWith(tp: Type)(using Context): Type = {
       assert(tp ne this, i"self instantiation of $origin, constraint = ${ctx.typerState.constraint}")
-      assert(!myInst.exists, i"$origin is already instantiated to $myInst but we attempted to instantiate it to $tp")
-      typr.println(i"instantiating $this with $tp")
+      if !myInst.exists then
+        typr.println(i"instantiating $this with $tp")
```

Comment:
Is removing the assert actually needed? The code doesn't seem to rely on instantiating the same variable multiple times (and if it did it would likely be fragile since `myInst` may or may not be set depending on the typerstate).

Reply:
In fact no, that was left in from an earlier attempt where the assertion triggered. I re-instantiated it.